src/find_file.rs

```diff
@@ -0,0 +1,109 @@
+use std::ffi::OsStr;
+use std::path::PathBuf;
+
+pub struct Criteria<'a> {
+    pub files: Vec<&'a str>,
+    pub extension: String,
+    pub folder: String,
+}
+
+// based on the directory do any of this criteria match or exist
+pub fn is_lang_project(dir_entry: &Vec<PathBuf>, criteria: &Criteria) -> bool {
+    dir_entry.into_iter().any(|path| match path.is_dir() {
+        true => has_folder(&path, &criteria.folder),
+        false => {
+            has_files(&path, &criteria.files)
+                || has_files_with_extension(&path, &criteria.extension)
+        }
+    })
+}
+
+pub fn has_files(dir_entry: &PathBuf, files: &Vec<&str>) -> bool {
+    let found_file = files.into_iter().find(|file| {
+        dir_entry
+            .file_name()
+            .and_then(OsStr::to_str)
+            .unwrap_or_default()
+            == **file
+    });
+
+    match found_file {
+        Some(file) => !file.is_empty(),
+        None => false,
+    }
+}
+
+pub fn has_files_with_extension(dir_entry: &PathBuf, extension: &String) -> bool {
```

Comment:
It looks to me like this is all the function is doing right now.

```suggestion
pub fn has_extension(dir_entry: &PathBuf, extension: &String) -> bool {
```

Reply:
👍